repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2980,57 +2977,123 @@ private void updateInConsistentOwnedMapVertices(AttributeMutationContext ctx, At
         }
     }
 
-    public void repairClassificationMappings(AtlasEntityHeader entityHeader, AtlasVertex entityVertex) throws AtlasBaseException {
-        if (entityHeader.getClassifications() != null) {
-            List<AtlasClassification> classifications = entityHeader.getClassifications();
-            String propagatedClassificationName = null;
-            String classificationNames = null;
-            String classificationText = null;
-            List<String> propagatedClassificationNamesList = new ArrayList<>();
-            List<String> classificationNamesList = new ArrayList<>();
-            for (AtlasClassification classification : classifications) {
-                String typeName = classification.getTypeName();
-                classificationText = fullTextMapperV2.getClassificationTextForEntity(new AtlasEntity(entityHeader));
-                if(classification.isPropagate() && !classification.getEntityGuid().equals(entityHeader.getGuid())) {
-                    //Add to propagatedClassificationNamesList
-                    propagatedClassificationNamesList.add(typeName);
-                    //Create delimiter separated string for propagatedClassificationNames
-                    if (StringUtils.isEmpty(propagatedClassificationName)) {
-                        propagatedClassificationName = CLASSIFICATION_NAME_DELIMITER + typeName + CLASSIFICATION_NAME_DELIMITER;
-                    } else {
-                        propagatedClassificationName = propagatedClassificationName + typeName + CLASSIFICATION_NAME_DELIMITER;
+    public void cleanUpClassificationPropagation(String classificationName) throws AtlasBaseException, IOException {
+        IndexSearchParams searchParams = new IndexSearchParams();
+        int batchSize = 100;
+        DirectIndexQueryResult indexQueryResult;
+        Map<String, Object> dsl = buildMap(classificationName);
+        searchParams.setDsl(dsl);
+        int totalCount = discoveryService.getQueryResponseCount(searchParams);
+        LOG.info("Total entities found for classification {} are {}", classificationName, totalCount);
+        int counter = 0;
+        dsl.put("from", 0);
```

Comment:
Why are we using elasticsearch for pagination? Why not use gremlin?

Reply:
It might lead to performance issues as the value from increases.

Reply:
- Re-evaluated the approach and it was very bad approach. Used JG to fetch attached vertices instead. This will also save us from inconsistency related to ES and Cassandra sync issue

Reply:
ES approach was also overly complicated